src/api/client/cache.ts

```diff
@@ -15,12 +15,38 @@ const cache = new InMemoryCache({
           const createdAtGte = args?.where?.createdAt_gte ? JSON.stringify(args.where.createdAt_gte) : ''
 
           // only for counting videos in HomeView
-          if (args?.where.channelId_in && !args?.first) {
+          if (args?.where?.channelId_in && !args?.first) {
             return `${createdAtGte}:${channelIdIn}`
           }
 
           return `${channelId}:${categoryId}:${channelIdIn}:${createdAtGte}`
         }),
+        videos: {
```

Comment:
Why aren't we using the helper from Apollo client?

Reply:
I'm not sure if they forgot it or what but on their function the read part is missing and coz of that it doesn't work properly, odd

Reply:
They didn't forget it :P It's the same as with cursor based pagination, the full set is returned always by design and you need to slice it yourself. I think we already discussed why this is

Reply:
I don't think I quite get it yet, maybe you can explain the idea on a quick call